diff -Nur cairo-1.12.2/configure cairo-1.12.2.n/configure
--- cairo-1.12.2/configure	2012-04-29 20:49:59.000000000 +0200
+++ cairo-1.12.2.n/configure	2013-05-31 23:58:18.000000000 +0200
@@ -32696,15 +32696,8 @@
 		cairo_cc_flag=no
 	fi
 
-	if test "x$cairo_cc_flag" = "xyes"; then
-		real_pthread_CFLAGS="$PTHREAD_CFLAGS";
-		 real_pthread_LIBS="$PTHREAD_LIBS";
-		 have_real_pthread=yes
-	else
 		real_pthread_CFLAGS="";
-		 real_pthread_LIBS="";
-		 have_real_pthread=no
-	fi
+		 have_real_pthread=yes
 	CFLAGS="$_save_cflags"
 	LDFLAGS="$_save_ldflags"
 	LIBS="$_save_libs"
