Only in cairo-1.10.2.new/: cairo-1.10.2
diff -ur cairo-1.10.2/configure cairo-1.10.2.new/configure
--- cairo-1.10.2/configure	2010-12-25 15:22:57.000000000 +0100
+++ cairo-1.10.2.new/configure	2011-03-27 13:28:55.000000000 +0200
@@ -32927,15 +32927,9 @@
 		cairo_cc_flag=no
 	fi
 
-	if test "x$cairo_cc_flag" = "xyes"; then
-		real_pthread_CFLAGS="$CFLAGS";
-		 real_pthread_LIBS="$LIBS";
-		 have_real_pthread=yes
-	else
-		real_pthread_CFLAGS="";
+     	real_pthread_CFLAGS="";
 		 real_pthread_LIBS="";
-		 have_real_pthread=no
-	fi
+		 have_real_pthread=yes
 	CFLAGS="$_save_cflags"
 	LDFLAGS="$_save_ldflags"
 	LIBS="$_save_libs"
